### ISSUE TITLE
Maybe ensure scalar api test tool prevents requests with "required" fields

### ISSUE THREAD

**Issue by FotieMConstant** [scalar/scalar#367]
### Expected Behavior

When using the Scalar api test tool, it is expected that it should prevent requests from going through if there are required fields specified as parameters in the request. The expected behavior is for Scalar to validate the request and block it when a required field is missing or not provided, ensuring that only valid and complete requests are sent to the server.

### Actual Behavior

Currently, Scalar behaves differently from the expected behavior. While it correctly identifies and indicates that a particular field is required, it does not prevent the request from being sent if the required field is missing or incomplete. This results in incomplete or incorrect requests being sent to the server.

### Reproduction steps

```bash
1. Use the example spec at: https://github.com/FotieMConstant/lanadoc/blob/main/public/resource/openapi-spec-sample.yaml
2. After setting up, Go to "Get a specific number of quotes" from the menu left
3. Click on test request
4. Click on send request
5. See that the request gets sent despite a required field "count"
```

### Version

0.8.2

### What browsers are you seeing the problem on?

Chrome

**Comment by hanspagel** [comment]
Great suggestion! Currently, the API client doesn’t know much about the spec. I think we’ll need to change this, this would open a lot of opportunities to improve the experience.

**Comment by FotieMConstant** [comment]
@hanspagel makes sense.

**Comment by hanspagel** [comment]
Yeah, I still think it should be possible to test it without required parameters. Generally speaking, the API client should know about the OpenAPI spec, but that’s a huge project on its own (which we’ll probably tackle soon). :)
